File: make_transparent.py

from PIL import Image
import numpy as np
import sys

def make_transparent(image_path, target_bg_color, tolerance=30):
    print(f"Processing {image_path}...")
    img = Image.open(image_path).convert("RGBA")
    data = np.array(img)

    # breakdown target color
    r_t, g_t, b_t = target_bg_color

    # breakdown image data
    r, g, b, a = data[:, :, 0], data[:, :, 1], data[:, :, 2], data[:, :, 3]

    # Calculate distance from target color
    # fast euclidean distance approximation
    mask = (np.abs(r - r_t) < tolerance) & (np.abs(g - g_t) < tolerance) & (np.abs(b - b_t) < tolerance)

    # Set alpha to 0 where mask is true
    data[mask, 3] = 0
    
    # Optional: Feathering/Anti-aliasing the edges could be done here but simple masking is a good start for flat backgrounds

    new_img = Image.fromarray(data)
    new_img.save(image_path)
    print(f"Saved transparent version to {image_path}")

# Dark Image (Black Background -> Transparent)
make_transparent(
    "/home/ussi/ai-learning-platform/frontend/public/images/practical-prompt-engineering_dark.png",
    (0, 0, 0),
    tolerance=15
)

# Light Image (White Background -> Transparent)
make_transparent(
    "/home/ussi/ai-learning-platform/frontend/public/images/practical-prompt-engineering_light.png",
    (255, 255, 255),
    tolerance=15
)

# Also update the default one (which is light)
make_transparent(
    "/home/ussi/ai-learning-platform/frontend/public/images/practical-prompt-engineering.png",
    (255, 255, 255),
    tolerance=15
)

File: make_transparent.js
const Jimp = require("jimp");
const path = require("path");

async function processImage(filename, targetColor, tolerance) {
  const activePath = path.join(__dirname, "frontend/public/images", filename);
  console.log(`Processing ${activePath}...`);

  try {
    const image = await Jimp.read(activePath);
    const targetR = targetColor[0];
    const targetG = targetColor[1];
    const targetB = targetColor[2];

    image.scan(
      0,
      0,
      image.bitmap.width,
      image.bitmap.height,
      function (x, y, idx) {
        const r = this.bitmap.data[idx + 0];
        const g = this.bitmap.data[idx + 1];
        const b = this.bitmap.data[idx + 2];

        if (
          Math.abs(r - targetR) <= tolerance &&
          Math.abs(g - targetG) <= tolerance &&
          Math.abs(b - targetB) <= tolerance
        ) {
          this.bitmap.data[idx + 3] = 0; // Set Alpha to 0
        }
      }
    );

    await image.writeAsync(activePath);
    console.log(`Success: ${filename}`);
  } catch (err) {
    console.error(`Error processing ${filename}:`, err);
  }
}

async function main() {
  // Dark Image (Black -> Transparent)
  await processImage("practical-prompt-engineering_dark.png", [0, 0, 0], 25);

  // Light Image (White -> Transparent)
  await processImage(
    "practical-prompt-engineering_light.png",
    [255, 255, 255],
    25
  );

  // Default (Light -> Transparent)
  await processImage("practical-prompt-engineering.png", [255, 255, 255], 25);
}

main();
